### PR TITLE
fetch: don't set an invalid origin header

### PR DESCRIPTION
**Pull request by KhafraDev**
refs: https://github.com/DIYgod/RSSHub/issues/15532

### CODE REVIEW

**Comment by KhafraDev** [comment]
this wasn't caught in the WPTs because we set a global origin which doesn't trigger the error case

File: lib/web/fetch/index.js [hunk 1]
```diff
@@ -444,8 +444,7 @@ function fetching ({
   // 9. If request’s origin is "client", then set request’s origin to request’s
   // client’s origin.
   if (request.origin === 'client') {
```
Comment (Uzlopak) [inline lib/web/fetch/index.js:446]:
```
  if (request.origin === 'client' && request.client?.origin) {
    request.origin = request.client.origin
  }
```

Comment (Uzlopak) [reply lib/web/fetch/index.js:446]: atleast a comment would be nice

**Review by Uzlopak — APPROVED** [review]
Makes sense

File: lib/web/fetch/util.js [hunk 2]
```diff
@@ -255,16 +255,23 @@ function appendFetchMetadata (httpRequest) {
 
 // https://fetch.spec.whatwg.org/#append-a-request-origin-header
 function appendRequestOriginHeader (request) {
-  // 1. Let serializedOrigin be the result of byte-serializing a request origin with request.
+  // 1. Let serializedOrigin be the result of byte-serializing a request origin
+  //    with request.
+  // TODO: implement "byte-serializing a request origin"
```
Comment (Uzlopak) [inline lib/web/fetch/util.js:260]: btw. you could use isomorphicEncode from lib/web/fetch/util.js

Comment (KhafraDev) [reply lib/web/fetch/util.js:260]: byte serializing a request origin is much more than isomorphic encoding

Comment (Uzlopak) [reply lib/web/fetch/util.js:260]: right

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**